zeppelin-zengine/src/test/java/org/apache/zeppelin/interpreter/launcher/SparkK8InterpreterLauncherTest.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.zeppelin.interpreter.launcher;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+import org.apache.zeppelin.conf.ZeppelinConfiguration;
+import org.apache.zeppelin.interpreter.InterpreterOption;
+import org.apache.zeppelin.interpreter.remote.SparkK8RemoteInterpreterManagedProcess;
+import org.junit.Test;
+
+import java.util.Properties;
+
+public class SparkK8InterpreterLauncherTest {
+
+  @Test
+  public void testLauncher() {
```

Comment:
Is it possible to add system test for K8SLauncher ?

Reply:
I'm not aware of such mini inmemory library for K8S as MiniHadoop, we can use mocks https://github.com/fabric8io/kubernetes-client/tree/master/kubernetes-server-mock to test Kubernetes call parts but this is not a full e2e system test.